internal/command/jobs/printers.go

```diff
@@ -110,6 +113,59 @@ func printJobDetails(cfg *config.Job, out io.Writer, job *models.Job) error {
 	return nil
 }
 
+func waitForJob(cfg *config.JobSubmit, out io.Writer, jobName string) error {
+
+	fmt.Fprintf(out, "Waiting for job execution\n")
+
+	looped := false
+
+	for {
+		j, err := getJob(cfg.Job, jobName)
+		if err != nil {
+			return err
+		}
+
+		switch j.Status.Phase {
+		case "completed":
+			os.Exit(int(j.Status.Attempts[0].State.Completed.ExitCode))
+
+			return nil
+		case "queued":
+			if looped {
+				fmt.Fprintf(out, "\033[1A\033[K")
+			}
+
+			fmt.Fprintf(out, "Queued on Job Runner Group %s\n", j.Spec.RunnerGroup)
```

Comment:
This code may cause a DoS! Note that while the job is in `queued` state the CLI will send requests to the `apiserver` continuously!
We should retry periodically.

Reply:
Right. Missed it during review.

Reply:
10 seconds before each retry or increasing retry??

Reply:
Yes, 10s sgtm.